api-lib/circleAdmin.ts

```diff
@@ -0,0 +1,36 @@
+import assert from 'assert';
+
+import type {
+  VercelRequest,
+  VercelResponse,
+  VercelApiHandler,
+} from '@vercel/node';
+import { z } from 'zod';
+
+import { getUserFromAuthHeader } from './findUser';
+import { verifyHasuraAdminMiddleware } from './validate';
+
+const middleware =
+  (handler: VercelApiHandler) =>
+  async (req: VercelRequest, res: VercelResponse) => {
+    // skip auth check if no header is passed
+    // because we can assume the user is a hasura-admin
+    if (req.headers.authorization) {
+      const circleId = z.number().parse(req.body.input.circle_id);
+      const { role } = await getUserFromAuthHeader(
+        req.headers.authorization,
+        circleId
+      );
+      if (role !== 1) {
+        return res.status(401).json({
+          message: 'User not circle admin',
+          code: 401,
+        });
+      }
+    }
+
+    await handler(req, res);
+  };
```

Comment:
@zashton here's the circle auth middleware

Reply:
instead of assuming it's admin, can we check the headers to verify it explicitly? Hasura should pass the auth/role info to the handler: https://hasura.io/docs/latest/graphql/core/remote-schemas/auth/remote-schema-forward-auth.html#passing-headers-from-hasura-to-your-remote-schema

Reply:
thank you! I was confused when I couldn't find these in the http headers.